secure/net/src/lib.rs

```diff
@@ -86,7 +86,8 @@ impl NetworkClient {
             let mut stream = TcpStream::connect(self.server);
 
             let sleeptime = time::Duration::from_millis(100);
-            while stream.is_err() {
+            while let Err(e) = stream {
+                debug!("Failed to connect to upstream {} {:?}", self.server, e);
```

Comment:
this probably would spam the logs, I'd recommend to try a few times and bail out

Reply:
It is worth considering what the right policy is here... I'm okay with this for now.

Reply:
I'm considering coming back and putting a timeout on these function.

The problem here is it will continue to poll forever, even if the destination is gone.  Rather error out, and handle in the outside logic how to find another server.

I couldn't test it failing on a unit test without either adding this or running a multi-threaded test that first starts with the service not existing, and then showing up.

Reply:
I personally wouldn't want to ship this version in that fashion, since we don't have the facilities for recovery in place just yet. If this failed out, we'd be stuck with a dead binary, which could be the result of one process starting later than another. Say I start Libra-node at time 0 and safety rules 2 minutes later. If our timeout window is 1 minute, the process would be dead :/.

Reply:
do what connectivity manager does and apply exp. backoff w/ a max delay.

https://github.com/libra/libra/blob/master/network/src/connectivity_manager/mod.rs#L522
https://github.com/libra/libra/blob/master/network/src/validator_network/network_builder.rs#L401

Reply:
I support that!